### ISSUE TITLE
Training loss equals to minimizing log(MSE)

### ISSUE THREAD

**Issue by yoyolicoris** [orybkin/sigma-vae#1]
Hi, have just read your paper, and I have some ideas want to share with.

In Eq. 5 from the paper, the ML estimation of variance equals to `MSE` loss. If we insert this into the objective function Eq. 3, the second term will always be `D/2`, which is a constant, and minimizing the objective means minimizing the first term `D * log(MSE) / 2`. 
This also equals to minimizing `MSE / (2 * (sigma ^ 2))` when using optimal sigma method because derivative of `log(x)` is `1 / x`.
So it probably means if we want an optimal  decoder, we should minimizing `log(MSE)` instead just `MSE`. (?

BTW, I also found your method is similiar to the following recent paper:
[Balancing reconstruction error and Kullback-Leibler divergence in Variational Autoencoders](https://arxiv.org/abs/2002.07514)

**Comment by orybkin** [comment]
Hi,

Indeed, the optimal loss is log(MSE)! This is a very funny fact, which we also realized at some point. However, the issue with this approach is that it computes the per-pixel variance, which does not work as well as a shared variance as we show in the paper. Therefore **I do not recommend using log(MSE) in practice**. The suggestion in this work is to use a shared variance, either trained with gradient descent, or just computed as the MSE loss.

The paper you linked seems to be based on "Diagnosing and Enhancing VAE Models", which already contains most of the ideas. "Diagnosing and Enhancing VAE Models" learns the variance, and indeed some other prior work does as well. We discuss this prior work in our paper. The contribution of our paper is an analysis of learning the variance, and practical suggestions on how to do it effectively, and what to avoid.

Oleg

**Comment by yoyolicoris** [comment]
I see. Thanks for your response and suggestions, will check the paper you mentioned.
